### PR TITLE
 Document the tectonicPuzzle example

### PR DESCRIPTION
**Pull request by pierre-guillou**
This PR adds a simple Python script and a mkdocs file to document the ttk-data tectonicPuzzle state file.

I did not understand fully the pipeline, the explanations may be a little terse.

Enjoy,
Pierre

### CODE REVIEW

File: python/tectonicPuzzle.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env python
+
+from paraview.simple import *
+
+# create a new 'XML Unstructured Grid Reader'
+tectonicPuzzlevtu = XMLUnstructuredGridReader(FileName=["tectonicPuzzle.vtu"])
+tectonicPuzzlevtu.PointArrayStatus = ["Viscosity"]
```
Comment (julien-tierny) [inline python/tectonicPuzzle.py]: I don't believe this line is necessary.

File: python/tectonicPuzzle.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+# create a new 'TTK TopologicalSimplification'
+tTKTopologicalSimplification1 = TTKTopologicalSimplification(
+    Domain=calculator1, Constraints=persistenceThreshold
+)
+tTKTopologicalSimplification1.ScalarField = ["POINTS", "logViscosity"]
+
+# create a new 'TTK MorseSmaleComplex'
+tTKMorseSmaleComplex1 = TTKMorseSmaleComplex(Input=tTKTopologicalSimplification1)
+tTKMorseSmaleComplex1.ScalarField = ["POINTS", "logViscosity"]
+tTKMorseSmaleComplex1.OffsetField = ["POINTS", "OutputOffsetScalarField"]
```
Comment (julien-tierny) [inline python/tectonicPuzzle.py]: I don't believe this line is necessary.

File: python/tectonicPuzzle.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+# create a new 'TTK TopologicalSimplification'
+tTKTopologicalSimplification2 = TTKTopologicalSimplification(
+    Domain=tTKTopologicalSimplification1, Constraints=pERSISTENT_MINIMA_AND_LARGE_MAXIMA
+)
+tTKTopologicalSimplification2.ScalarField = ["POINTS", "logViscosity"]
+
+# create a new 'TTK PersistenceDiagram'
+tTKPersistenceDiagram2 = TTKPersistenceDiagram(Input=tTKTopologicalSimplification2)
+tTKPersistenceDiagram2.ScalarField = ["POINTS", "logViscosity"]
+tTKPersistenceDiagram2.InputOffsetField = ["POINTS", "OutputOffsetScalarField"]
```
Comment (julien-tierny) [inline python/tectonicPuzzle.py]: I don't believe this line is necessary.

File: python/tectonicPuzzle.py [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+tTKTopologicalSimplification3 = TTKTopologicalSimplification(
+    Domain=tTKTopologicalSimplification2,
+    Constraints=lARGE_MAXIMA_LOW_SADDLE_AND_PERSISTENT_MINIMA,
+)
+tTKTopologicalSimplification3.ScalarField = ["POINTS", "logViscosity"]
+
+# create a new 'TTK MorseSmaleComplex'
+tTKMorseSmaleComplex2 = TTKMorseSmaleComplex(Input=tTKTopologicalSimplification3)
+tTKMorseSmaleComplex2.ScalarField = ["POINTS", "logViscosity"]
+tTKMorseSmaleComplex2.OffsetField = ["POINTS", "OutputOffsetScalarField"]
```
Comment (julien-tierny) [inline python/tectonicPuzzle.py]: I don't believe this line is necessary.

**Review by julien-tierny — CHANGES_REQUESTED** [review]
thanks a lot for the PR.
Could you please address the few minor remarks I entered?
thanks!

**Comment by pierre-guillou** [comment]
Those lines have been removed

**Review by julien-tierny — APPROVED** [review]

**Comment by julien-tierny** [comment]
awesome, thanks!

**Pull request merged**